### DISCUSSION TITLE
jsr305 and jakarta

### DISCUSSION THREAD

**Discussion by wrestang** [google/guice#1746] [Q&A] [answered]
Guice version 6.0.0 allows for the slow transition away from javax classes, and while javax.inject, javax.servlet, and javax.persistence are mentioned by name, javax.annotation is not.  According to [the wiki](https://github.com/google/guice/wiki/UseNullable) we should still be using javax.annotation.  Is there any plan to migrate to jakarta for this? Bonus question that I could do work to find out, but figured I'd get an official word on instead: is there any current support for jakarta.annotation over the findbugs/jsr305 jar as of Guice 6.0.0/7.0.0?

**Comment by sameb** [comment, marked as answer]
Per https://github.com/google/guice/wiki/UseNullable#supported-nullable-annotations, Guice recognizes any annotation whose simple name is "Nullable". Guice has no preference for the javax nor the jakarta namespace for Nullable annotations. The wiki was just written when the javax one was the main/only option. I'll update the wiki to lead with the jakarta namespace instead.

**Reply by wrestang** [reply]
Thanks! I was speed reading through that part and just didn't pick up on that!

**Reply by sameb** [reply]
(Wiki updates are published now.)
